### PR TITLE
Secret help - creating .dockercfg from file should be done via 'oc secrets new'

### PR DESCRIPTION
**Pull request by jhadvig**
@liggitt PTAL
https://github.com/openshift/origin/issues/4430

### CODE REVIEW

File: pkg/cmd/cli/secrets/dockercfg.go [hunk 1]
```diff
@@ -37,9 +37,9 @@ If you don't already have a .dockercfg file, you can create a dockercfg secret d
 
   $ %s SECRET_NAME --docker-server=DOCKER_REGISTRY_SERVER --docker-username=DOCKER_USER --docker-password=DOCKER_PASSWORD --docker-email=DOCKER_EMAIL
 
-If you do already have a .dockercfg file, you can create a dockercfg secret by using:
+If you do already have a .dockercfg file, you can create a dockercfg secret by using the 'oc secrets new' command:
 
-  $ %s SECRET_NAME path/to/.dockercfg
+  $ oc secrets new SECRET_NAME path/to/.dockercfg
```
Comment (mfojtik) [inline pkg/cmd/cli/secrets/dockercfg.go]: %s

Comment (liggitt) [inline pkg/cmd/cli/secrets/dockercfg.go]: @mfojtik is right, we can't hardcode command names, but `%s` will give `oc secrets new-dockercfg`, which is not the example we want... we need another token for the recommended `oc secrets new` command string

Comment (jhadvig) [inline pkg/cmd/cli/secrets/dockercfg.go]: @liggitt I dont follow, what another token do you mean ?

Comment (liggitt) [inline pkg/cmd/cli/secrets/dockercfg.go]: another format arg, sorry

Comment (jhadvig) [inline pkg/cmd/cli/secrets/dockercfg.go]: is there a way how to do it now ?

Comment (liggitt) [inline pkg/cmd/cli/secrets/dockercfg.go]: pass in the other command, like `NewCmdRequestProject` takes a login command

**Comment by jhadvig** [comment]
@mfojtik @liggitt comment addressed. PTAL

**Comment by jhadvig** [comment]
[test]

File: pkg/cmd/cli/secrets/dockercfg.go [hunk 2]
```diff
@@ -37,9 +37,9 @@ If you don't already have a .dockercfg file, you can create a dockercfg secret d
 
   $ %s SECRET_NAME --docker-server=DOCKER_REGISTRY_SERVER --docker-username=DOCKER_USER --docker-password=DOCKER_PASSWORD --docker-email=DOCKER_EMAIL
 
-If you do already have a .dockercfg file, you can create a dockercfg secret by using:
+If you do already have a .dockercfg file, you can create a dockercfg secret by using the 'oc secrets new' command:
 
-  $ %s SECRET_NAME path/to/.dockercfg
+  $ %[2]s SECRET_NAME path/to/.dockercfg
```
Comment (liggitt) [inline pkg/cmd/cli/secrets/dockercfg.go]: you need to actually pass in a second Sprintf arg to use %2

Comment (jhadvig) [inline pkg/cmd/cli/secrets/dockercfg.go]: @liggitt I dont follow. The %[2]s will properly return `newSecretFullName` which contains `oc secrets new` string.
Or whats the issue ?

Comment (liggitt) [inline pkg/cmd/cli/secrets/dockercfg.go]: oh, didn't notice we were already passing in a second arg for the new secret command. in that case, switch all the `%s` in the format string to `%[n]s` style

Comment (jhadvig) [inline pkg/cmd/cli/secrets/dockercfg.go]: @liggitt done :+1:

File: pkg/cmd/cli/secrets/dockercfg.go [hunk 3]
```diff
@@ -35,11 +35,11 @@ by creating a dockercfg secret and attaching it to your service account.
 
 If you don't already have a .dockercfg file, you can create a dockercfg secret directly by using:
 
-  $ %s SECRET_NAME --docker-server=DOCKER_REGISTRY_SERVER --docker-username=DOCKER_USER --docker-password=DOCKER_PASSWORD --docker-email=DOCKER_EMAIL
+  $ %[1]s SECRET_NAME --docker-server=DOCKER_REGISTRY_SERVER --docker-username=DOCKER_USER --docker-password=DOCKER_PASSWORD --docker-email=DOCKER_EMAIL
 
-If you do already have a .dockercfg file, you can create a dockercfg secret by using:
+If you do already have a .dockercfg file, you can create a dockercfg secret by using the 'oc secrets new' command:
```
Comment (liggitt) [inline pkg/cmd/cli/secrets/dockercfg.go]: `by using:` (don't hardcode the command name here)

File: pkg/cmd/cli/secrets/dockercfg.go (hunk 2 above)
Comment (liggitt) [inline pkg/cmd/cli/secrets/dockercfg.go]: missed one below

Comment (jhadvig) [inline pkg/cmd/cli/secrets/dockercfg.go]: @liggitt ups, didnt notices that one, was looking only on the line separated examples.. sorry
anyways its there :)

**Comment by liggitt** [comment]
[merge]

**Comment by mfojtik** [comment]
re - [merge]

**Pull request merged**